aerleon/lib/openconfig.py

```diff
@@ -196,17 +187,52 @@ def ConvertToDict(
                                         raise OcFirewallError(
                                             'Protocol %s unknown. Use an integer.', proto
                                         )
-                                    ace_dict[family]['config']['protocol'] = proto_num
+                                    self.SetProtocol(ace_dict, family, proto_num)
                                 rules.append(copy.deepcopy(ace_dict))
                             else:
-                                proto_num = proto
-                                ace_dict[family]['config']['protocol'] = proto_num
+                                self.SetProtocol(ace_dict, family, proto)
                                 # This is the business end of ace explosion.
-                                # A dict is a reference type, so deepcopy is atually required.
+                                # A dict is a reference type, so deepcopy is actually required.
                                 rules.append(copy.deepcopy(ace_dict))
 
         return rules
 
+    def SetAction(self, dict: dict) -> None:
```

Comment:
A lot of these functions are being called `SetFoo` but are not setting something on the class but on some passed in object. SetAction is likely not even a public function. If someone makes a Openconfig object are they expected to ever call SetAction? You may wish to consider assigning the dict to the object and operating it itself.

Reply:
added ```term_dict``` as a member